src/base-apis.js

```diff
@@ -277,7 +277,21 @@ MatrixBaseApis.prototype.loginWithPassword = function(user, password, callback)
     return this.login("m.login.password", {
```

Comment:
Should we be doing this in the `login` method instead to keep the API somewhat consistent in between these two methods?

Reply:
Good point. As discussed, I've refactored the code to the base 'login' method.